### DISCUSSION TITLE
Error on updating a file

### DISCUSSION THREAD

**Discussion by ahmedk92** [octokit/octokit.js#1809] [Q&A] [answered]
I'm using this code to get the latest `sha` to update `README.md`:

```javascript
const commit = await octokit.repos.getCommit({
                  owner: context.repo.owner,
                  repo: context.repo.repo,
                  ref: 'master'
               });
octokit.repos.createOrUpdateFileContents({
  sha: commit.data.sha,
  owner: context.repo.owner,
  repo: context.repo.repo,
  path: 'README.md',
  message: 'Update README',
  content: 'VGVzdCB0ZXN0',
  commiter: {
  name: 'update-readme-bot', 
  email: 'example@example.com'
  },
  author: {
    name: 'update-readme-bot',
    email: 'example@example.com' 
  }
});
```

But I keep getting this error:
```
Unhandled error: HttpError: README.md does not match <sha in commit.data.sha>
```

What `sha` should I get? and how to get it?

Thanks.

**Comment by gr2m** [comment, marked as answer]
What you want is the file sha, not the commit sha. You can get the file sha using `octokit.repos.getContents({owner, repo, path})`. The response data includes `sha`, which is the sha you need to set when updating the file.

**Reply by gr2m** [reply]
In the case of the repository's README, you can also use `octokit.repos.getReadme({ owner, repo })`. The response includes `sha` and also `path` which you can then use with `octokit.repos.createOrUpdateFileContents`, as different file names can be used for repository READMES, such as `README`, `README.md` or `readme.md`

**Reply by ahmedk92** [reply]
That worked! That you for the quick reply!
